lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -1156,6 +1239,18 @@ llvm::Function *CFEPAnalyzer<FO>::createFakeFunction(llvm::BasicBlock *Entry) {
   return FakeFunction.extractFunction();
 }
 
+template<class FO>
+void CFEPAnalyzer<FO>::initializeFakeFunctions() {
+  for (const auto &Function : Binary.Functions) {
+    if (Function.Type != model::FunctionType::Fake)
+      continue;
+
+    auto &Summary = Oracle.at(Function.Entry);
```

Comment:
Uhm, I'd expect the Oracle not to have an entry associated to `Function.Entry` and `at()` should assert in such case.

Reply:
Why do you say? The Oracle should have an entry associated to that `Function.Entry` for sure at this point since the model has already been imported.

Reply:
ACK. If it's already there you don't need to set `Summary.Type = model::FunctionType::Fake`.
I'd avoid touching the model entirely in this function, you can iterate over each function present in the `Oracle` and, if it has `.Type == Fake`, then call `createFakeFunction`.

Otherwise this a bit confusing.

Reply:
I think that setting `Summary.Type` (within the local cache of EFA, not the model) is necessary here. In fact, we need to query the Oracle (via `isFakeFunction`, which checks `Summary.Type`) when re-scheduling an analysis for the fake function callers, if we got improved results for a specific function (recall that, when inserting the functions to be analyzed into the queue, we need to recompute its callers, if a caller turns out to be fake).

Reply:
Removed, fixed in `importModel`.